src/Persistence/Sql/Query.php

```diff
@@ -0,0 +1,291 @@
+    }
+
+    public function where($fieldName, $operator = null, $value = null): AbstractQuery
+    {
+        $this->fillWhere($this->dsql, new Model\Scope\Condition(...func_get_args()));
+
+        return $this;
+    }
+
+    protected function initOrder(): void
+    {
+        $this->dsql->reset('order');
+
+        foreach ((array) $this->order as [$field, $desc]) {
+            if (is_string($field)) {
+                $field = $this->model->getField($field);
+            }
+
+            if (!$field instanceof Expression && !$field instanceof Expressionable) {
+                throw (new Exception('Unsupported order parameter'))
+                    ->addMoreInfo('model', $this->model)
+                    ->addMoreInfo('field', $field);
+            }
+
+            $this->dsql->order($field, $desc);
+        }
+    }
+
+    protected function initLimit(): void
+    {
+        $this->dsql->reset('limit');
+
+        if ($args = $this->getLimitArgs()) {
+            $this->dsql->limit(...$args);
+        }
+    }
+
+    protected function doExecute()
+    {
+        return $this->dsql->execute();
+    }
+
+    protected function doGet(): array
+    {
+        return $this->dsql->get();
+    }
+
+    protected function doGetRow(): ?array
+    {
+        return $this->dsql->getRow();
+    }
+
+    protected function doGetOne()
+    {
+        return $this->dsql->getOne();
+    }
+
+    public function getDsqlExpression($expression = null)
+    {
+        return $this->dsql;
+    }
+
+    /**
+     * Return the underlying Dsql object performing the query to DB.
+     *
+     * @return \atk4\dsql\Query
+     */
+    public function dsql()
+    {
+        return $this->dsql;
+    }
+
+    protected function initWhere(): void
+    {
+        $this->fillWhere($this->dsql, $this->scope);
+    }
+
+    protected static function fillWhere(DsqlQuery $query, Model\Scope\AbstractScope $condition)
+    {
+        if (!$condition->isEmpty()) {
+            // peel off the single nested scopes to convert (((field = value))) to field = value
+            $condition = $condition->simplify();
+
+            // simple condition
+            if ($condition instanceof Model\Scope\Condition) {
+                $query->where(...$condition->toQueryArguments());
+            }
+
+            // nested conditions
+            if ($condition instanceof Model\Scope) {
+                $expression = $condition->isOr() ? $query->orExpr() : $query->andExpr();
+
+                foreach ($condition->getNestedConditions() as $nestedCondition) {
+                    self::fillWhere($expression, $nestedCondition);
+                }
+
+                $query->where($expression);
+            }
+        }
+    }
+
+    public function getDebug(): array
+    {
+        return array_merge([
+            'sql' => $this->dsql->getDebugQuery(),
+        ], parent::getDebug());
+    }
+
+    public function __call($method, $args)
```

Comment:
Better to declare the 3 methods manually and abstract the impl completely

Reply:
Too many of the underlying DSQL methods are used
I can expand the annotation

Reply:
if like 5 now keep it for now - but do we really want to offer all of these in the data? that is the main question...

Reply:
Later we can consider deprecating this but let's keep this direct forward for now.

Reply:
more about better abstraction, code otherwise clean, fine with this now